app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/ResourceState.kt

```diff
@@ -0,0 +1,13 @@
+package com.kks.nimblesurveyjetpackcompose.model
+
+sealed class ResourceState<out T> {
```

Comment:
Any reason why we have so many states here? Wouldn't a simple `Success`, `Error` and maybe `Loading` be enough? 🤔

Reply:
In order to be able to handle different kinds of `Error` in different ways. If I only have `Error`, I cannot know is it because of no internet connection or is it because `HttpException`

Reply:
Or have `Error` take a `Throwable` instead of a `String` and extract the error from the throwable's type 🤔 
Something like 
```kotlin
class Error(val throwable: Throwable): ResourceState<Nothing>()

// Domain level exception class so that we don't have to check for things like `HttpException` on layers that don't need it
class NetworkError(val code: Int): Exception()

return ResourceState.Error(NetworkError(404))
```

CC: @AVI5HEK thoughts? 🤔

Reply:
I like @ryan-conway 's idea. Also, choosing the appropriate message string is a presentation logic and should belong to the Activity/Fragment.

Reply:
Updated the class 🚀  @ryan-conway